ring/basic_lifecycler_delegates.go

```diff
@@ -164,15 +164,15 @@ func NewInstanceRegisterDelegate(state InstanceState, tokenCount int) InstanceRe
 	}
 }
 
-func (d InstanceRegisterDelegate) OnRingInstanceRegister(_ *BasicLifecycler, ringDesc Desc, instanceExists bool, _ string, instanceDesc InstanceDesc) (InstanceState, Tokens) {
+func (d InstanceRegisterDelegate) OnRingInstanceRegister(l *BasicLifecycler, ringDesc Desc, instanceExists bool, _ string, instanceDesc InstanceDesc) (InstanceState, Tokens) {
 	// Keep the existing tokens if any, otherwise start with a clean situation.
 	var tokens []uint32
 	if instanceExists {
 		tokens = instanceDesc.GetTokens()
 	}
 
 	takenTokens := ringDesc.GetTokens()
-	newTokens := GenerateTokens(d.tokenCount-len(tokens), takenTokens)
+	newTokens := l.tokenGenerator.GenerateTokens(d.tokenCount-len(tokens), takenTokens)
```

Comment:
Projects can implement their own delegates, but will not have access to `tokenGenerator`. I think it's better if you expose `BasicLifecycler.TokenGenerator()` (as a function so that it's read-only), so that other projects can access it too.